data/labels.go

```diff
@@ -71,18 +71,26 @@ func (l Labels) String() string {
 
 // LabelsFromString parses the output of Labels.String() into
 // a Labels object. It probably has some flaws.
+// This should accommodate different styles of input like prometheus, influx
```

Comment:
you stand by the "It probably has some flaws" part?  

Can you read through the comments in this file and see if the things that shout "this might not work" either do or do not work (now that we have tests)?